src/debug/Stream.h

```diff
@@ -207,6 +207,11 @@ void ResyncDebugLog(FILE *newDestination);
         } \
    } while (/*CONSTCOND*/ 0)
 
+/// Convenience wrapper for displaying protocol message flow in cache.log
+#define traceProtocol(SECTION, LABEL, ID, MESSAGE) \
+    debugs(SECTION, 2, LABEL << ": " << ID << \
+           "\n----------\n" << MESSAGE << "\n----------")
```

Comment:
When dealing with boundary separators, it would be really nice to switch from very inconvenient dash counting/typing to words that are easy to type and search for. A few leading dashes is more than enough to make these strings unique (because Squid virtually never prints dashes at the beginning of a line) and we can add a few trailing dashes to decorate (they will not be needed when searching for things). For example:

```suggestion
           "\n--begin--\n" << MESSAGE << "\n--end----")
```

or 

```suggestion
           "\n--start--\n" << MESSAGE << "\n--stop---")
```